Client/Coordinators/Browser/BrowserCoordinator.swift

```diff
@@ -149,4 +149,28 @@ class BrowserCoordinator: BaseCoordinator, LaunchCoordinatorDelegate, BrowserDel
             return false
         }
     }
+
+    private func handle(query: String) {
+        browserViewController.openBlankNewTab(focusLocationField: false)
+        browserViewController.urlBar(browserViewController.urlBar, didSubmitText: query)
+    }
+
+    private func handle(url: URL?, isPrivate: Bool, options: Set<Route.SearchOptions>? = nil) {
+        if let url = url {
+            if options?.contains(.switchToNormalMode) == true {
+                browserViewController.switchToPrivacyMode(isPrivate: false)
+            }
+            browserViewController.switchToTabForURLOrOpen(url, isPrivate: isPrivate)
+        } else {
+            browserViewController.openBlankNewTab(focusLocationField: options?.contains(.focusLocationField) == true, isPrivate: isPrivate)
+        }
+    }
+
+    private func handle(searchURL: URL?, tabId: String) {
+        if let url = searchURL {
+            browserViewController.switchToTabForURLOrOpen(url, uuid: tabId, isPrivate: false)
+        } else {
+            browserViewController.openBlankNewTab(focusLocationField: true, isPrivate: false)
+        }
+    }
```

Comment:
Im wondering if we should put this logic in BVC instead. @OrlaM what do you think?

Reply:
I assume the long plan is to remove logic like this from bvc. I'm not sure if it helps to move it there.

Reply:
Had a chat about this with Orla; The long term goal is indeed to remove logic from BVC, but this logic shouldn't live in the coordinator for the sake of removing it from BVC since coordinators should only handle the flow of navigation. In an ideal world we would have a better place to put this already, but it's out of scope for this project and will be handled later on. 

So for now we should just have methods that we call directly on BVC to pass the route arguments, and let BVC handle the logic. Example: in the switch case we would call `handle(url: URL?, isPrivate: Bool, options: Set<Route.SearchOptions>? = nil)` on BVC, and BVC would then unwrap the URL, check for options and so on.